src/main/java/examples/ExampleIngestFunction3.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+package examples;
+
+import com.salesforce.functions.jvm.sdk.Context;
+import com.salesforce.functions.jvm.sdk.InvocationEvent;
+import com.salesforce.functions.jvm.sdk.SalesforceFunction;
+import com.salesforce.functions.jvm.sdk.data.bulk.*;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+
+public class ExampleIngestFunction3 implements SalesforceFunction<String, List<String>> {
+  @Override
+  public List<String> apply(InvocationEvent<String> event, Context context) throws Exception {
+    BulkApi bulkApi = context.getOrg().get().getBulkDataApi();
+
+    // This example demonstrates adding arbitrary objects as row data. This is common when a
+    // function is used to ingest data from an external system. In this example the values are
+    // hardcoded, but could
+    // easily come from an external system via a database connection or HTTP request.
+    ExternalAccountObject externalAccountObject1 =
+        new ExternalAccountObject("Harald Hollister", "Harald manages the toy division", 43);
+    ExternalAccountObject externalAccountObject2 =
+        new ExternalAccountObject(
+            "Getchen Gerstlauer", "Product owner for the vending machine software", 51);
+    ExternalAccountObject externalAccountObject3 =
+        new ExternalAccountObject("Janine Sparks", "3D printing influencer", 26);
+
+    // This is the mapper we use to transform the external objects into rows. See the class
+    // implementation for details.
+    DataTableFieldValueExtractor<ExternalAccountObject> mapper =
+        new MyCustomRowDataTableFieldValueExtractor();
+
+    DataTable table =
+        bulkApi
+            .createDataTableBuilder("FirstName", "LastName")
+            .addRow(externalAccountObject1, mapper)
+            .addRow(externalAccountObject2, mapper)
+            .addRow(externalAccountObject3, mapper)
+            .build();
+
+    // In this example, we just "fire and forget" and return the bulk ingest ids
+    return bulkApi.createSimpleIngestJob("Account", IngestOperation.INSERT, table).stream()
+        .map(IngestJobReference::getId)
+        .collect(Collectors.toList());
+  }
+
+  private static class MyCustomRowDataTableFieldValueExtractor
+      implements DataTableFieldValueExtractor<ExternalAccountObject> {
+    @Nonnull
+    @Override
+    public Optional<String> extractFieldValue(ExternalAccountObject data, String fieldName) {
```

Comment:
would you consider adding an `index` value into this mapper's signature?  actually, can you even do that with java's functions?